### DISCUSSION TITLE
Separating Authentication from Authorization

### DISCUSSION THREAD

**Discussion by markdav-is** [oqtane/oqtane.framework#910] [Q&A]
We have an existing ASP.Net Identity database with AspNetUsers etc.  We'd like to have our folks login there (authenticate) and then have Oqtane maintain the authorization using it's Users table, granular authentication, etc.  Currently, these use the same database connection.  

one option would be to introduce an AuthConnection to each Tenant.  This would allow a place to set this and use the existing sign-in, etc.

another option would be for us to create our own auth module, and stop using the provided login controls.  If we did that, we might be able to sync the AspNetUsers tables between Oqtane tenant and our auth database.

Has anyone rolled a separate auth from the OOTB auth?

**Comment by sbwalker** [comment]
This is similar to #903 - as they both relate to overriding default services in Oqtane.

**Reply by hishamco** [reply]
I will try to prototype this in the upcoming days

**Reply by sbwalker** [reply]
@hishamco did you see my comment in that thread? "I have another suggestion... rather than introduce more interfaces ( which will impose a performance hit on startup ) why not modify the core framework to use the existing IServerStartup interface to dynamically register services which can be replaced? This is the whole purpose of IServerStartup and if we expect modules to use it then it makes sense that the core framework should use it as well ( as this provides a functional implementation ). Basically we would create a number of granular implementations of the IServerStartup interface for various services that are currently hardcoded in main Startup class, and include feature flags to enable/disable them. Then if a third party developer wanted to register their own Identity service configuration they could implement IServerStartup and disable the default Identity service registration."

**Reply by markdav-is** [reply]
I really like the feature-flags idea.  maybe having a "default services" optional section in the appsettings, would be a handy way to turn things off so they could be overwritten.
